tests/snapshots/snapshots__collatz.msl.snap

```diff
@@ -0,0 +1,39 @@
+---
+source: tests/snapshots.rs
+expression: msl
+---
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+typedef metal::uint3 type;
+typedef uint type1;
+typedef type1 type2[1];
+struct PrimeIndices {
+	type2 data;
+};
+
+type1 collatz_iterations(
+	type1 n
+) {
+	type1 i;
+	i = static_cast<uint>(0);
+	while(true) {
+		if (n <= static_cast<uint>(1)) {
+			break;
+		}
+		if (n % static_cast<uint>(2) == static_cast<uint>(0)) {
+			n = n / static_cast<uint>(2);
+		} else {
+			n = static_cast<uint>(3) * n + static_cast<int>(1);
+		}
+		i = i + static_cast<uint>(1);
+	}
+	return i;
+}
+kernel void main1(
```

Comment:
Just making sure as I have no knowledge about Metal, but is Metal kernel not the same as SPIR-V kernel? If true, my follow up question would be: is this allowed to be used?

Reply:
Correct, in metal the compute entry points are called "kernel".